src/components/FiltersButton/FiltersButton.tsx

```diff
@@ -33,16 +33,22 @@ export const FiltersButton = forwardRef<HTMLButtonElement, FiltersButtonProps>(
   ({ className, hasSelectedFilters, triggerText, ...other }, ref) => {
     const componentClassName = clsx(styles['filters-button'], className);
 
-    const buttonVariant = hasSelectedFilters ? 'primary' : 'secondary';
-    const buttonStatus = hasSelectedFilters ? 'brand' : 'neutral';
+    const variantStatus: VariantStatus = hasSelectedFilters
+      ? {
+          variant: 'primary',
+          status: 'brand',
+        }
+      : {
+          variant: 'secondary',
+          status: 'neutral',
+        };
 
     return (
       <Button
         className={componentClassName}
         ref={ref}
-        status={buttonStatus}
-        variant={buttonVariant}
         {...other}
+        {...variantStatus}
```

Comment:
apply after `{...other}` to make sure any values in rest are replaced with those calculated in this component directly.